src/main/java/software/amazon/msk/auth/iam/internals/MSKCredentialProvider.java

```diff
@@ -267,6 +268,22 @@ public int getMaxBackOffTimeMs() {
                     .orElse(DEFAULT_MAX_BACK_OFF_TIME_MS);
         }
 
+        public EndpointConfiguration buildEndpointConfiguration(String stsRegion){
+            //An AWSSecurityTokenService with a regional endpoint configuration
+            EndpointConfiguration endpointConfiguration =
+                    new EndpointConfiguration(
+                            String.format("sts.%s.amazonaws.com", stsRegion),
+                            stsRegion);
```

Comment:
Thanks for the PR. 
When you follow https://docs.aws.amazon.com/sdkref/latest/guide/feature-sts-regionalized-endpoints.html
and set
```
[default]
sts_regional_endpoints = regional
```
doesn't it pick it up by itself?

Reply:
I'm running my job on KDA and I can't do such a setup on this service.